src/rime/resource.cc

```diff
@@ -27,19 +31,48 @@ string ResourceResolver::ToFilePath(const string& resource_id) const {
          (missing_suffix ? type_.suffix : "");
 }
 
+#ifdef _MSC_VER
+static std::string U16ToACP(const std::wstring& wstr) {
+  std::string ret;
+  int length = static_cast<int>(wstr.length());
+  int convcnt =
+      WideCharToMultiByte(CP_ACP, 0, wstr.c_str(), length, NULL, 0, NULL, NULL);
+  if (convcnt > 0) {
+    ret.resize(convcnt);
+    WideCharToMultiByte(CP_ACP, 0, wstr.c_str(), length, &ret[0], convcnt, NULL,
+                        NULL);
+  }
+  return ret;
+}
+#endif
+
 std::filesystem::path ResourceResolver::ResolvePath(const string& resource_id) {
+#ifdef _MSC_VER
+  std::wstring platform_id_wstr =
+      opencc::UTF8Util::GetPlatformString(resource_id);
+  std::string resource_id_ = U16ToACP(platform_id_wstr);
+#else
+  std::string resource_id_ = resource_id;
+#endif
   return std::filesystem::absolute(
       root_path_ /
-      std::filesystem::path(type_.prefix + resource_id + type_.suffix));
+      std::filesystem::path(type_.prefix + resource_id_ + type_.suffix));
 }
 
 std::filesystem::path FallbackResourceResolver::ResolvePath(
     const string& resource_id) {
-  auto default_path = ResourceResolver::ResolvePath(resource_id);
+#ifdef _MSC_VER
+  std::wstring platform_id_wstr =
+      opencc::UTF8Util::GetPlatformString(resource_id);
+  std::string resource_id_ = U16ToACP(platform_id_wstr);
```

Comment:
I dislike this fix.

maybe we can use `filesystem::path::native` to get Windows local filesystem encoding.

Reply:
> I dislike this fix.
> 
> maybe we can use `filesystem::path::native` to get Windows local filesystem encoding.

as far as I've known, native seems not for that? 
with c++17 standard std::filesystem::u8path(utf8String) may be a solution. however it's deprecated in c++20...

Reply:
u8path as the solution